devicetypes/smartthings/zigbee-window-shade.src/zigbee-window-shade.groovy

```diff
@@ -122,8 +122,9 @@ def parse(String description) {
 def levelEventHandler(currentLevel) {
 	def lastLevel = device.currentValue("level")
 	log.debug "levelEventHandle - currentLevel: ${currentLevel} lastLevel: ${lastLevel}"
-	if (lastLevel == "undefined" || currentLevel == lastLevel) { //Ignore invalid reports
+	if ((lastLevel == "undefined" || currentLevel == lastLevel) && state.invalidSameLevelEvent) { //Ignore invalid reports
 		log.debug "Ignore invalid reports"
+		state.invalidSameLevelEvent = true
```

Comment:
Could You tell why do invalidSameLevelEvent is set to true here if the above condition already checks it ?
Did You mean to check first if that state variable is defined ? (if it will be defined and set to false then the whole condition might not be met anyway).

Reply:
Oh yeah, thanks. It was meant for `else { ... }` block.